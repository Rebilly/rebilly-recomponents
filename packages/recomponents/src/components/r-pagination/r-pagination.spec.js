import {shallowMount} from '@vue/test-utils';
import {renderToString} from '@vue/server-test-utils';
import RPagination from './r-pagination.vue';

describe('r-pagination.vue', () => {
    it('should render Wrapper and match snapshot', () => {
        const wrapper = shallowMount(RPagination, {
            propsData: {
                provider: () => {
                },
                total: 10,
                limit: 5,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should render via SSR and match snapshot', async () => {
        const wrapper = renderToString(RPagination, {
            propsData: {
                provider: () => {
                },
                total: 10,
                limit: 5,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should enable hasPagination property when limit < total', () => {
        const wrapper = shallowMount(RPagination, {
            propsData: {
                provider: () => {
                },
                total: 10,
                limit: 5,
            },
        });

        expect(wrapper.vm.hasPagination).toEqual(true);
    });

    it('should disable hasPagination property when limit < total', () => {
        const wrapper = shallowMount(RPagination, {
            propsData: {
                provider: () => {
                },
                total: 3,
                limit: 5,
            },
        });

        expect(wrapper.vm.hasPagination).toEqual(false);
    });

    it('pages count correctly when total/limit < 1.5', () => {
        const wrapper = shallowMount(RPagination, {
            propsData: {
                provider: () => {
                },
                total: 5,
                limit: 4,
            },
        });

        expect(wrapper.vm.pages).toEqual(2);
    });

    it('pages count correctly when total/limit > 1.5', () => {
        const wrapper = shallowMount(RPagination, {
            propsData: {
                provider: () => {
                },
                total: 5,
                limit: 2,
            },
        });

        expect(wrapper.vm.pages).toEqual(3);
    });

    it('hasNext property correctly', () => {
        const wrapper = shallowMount(RPagination, {
            propsData: {
                provider: () => {
                },
                total: 5,
                limit: 2,
            },
        });

        expect(wrapper.vm.hasNext).toEqual(true);
        wrapper.setData({page: 3});
        expect(wrapper.vm.hasNext).toEqual(false);
    });

    it('hasPrevious property correctly', () => {
        const wrapper = shallowMount(RPagination, {
            propsData: {
                provider: () => {
                },
                total: 5,
                limit: 2,
            },
        });

        expect(wrapper.vm.hasPrevious).toEqual(false);
        wrapper.setData({page: 3});
        expect(wrapper.vm.hasPrevious).toEqual(true);
    });

    it('items property correctly', () => {
        const wrapper = shallowMount(RPagination, {
            propsData: {
                provider: () => {
                },
                total: 100,
                limit: 10,
                totalVisible: 5,
            },
        });

        const getVisible = (items) => Object.values(items).map(({content}) => content);
        expect(getVisible(wrapper.vm.items)).toEqual([1, 2, 3, 4, 5, undefined]);
        wrapper.setData({page: 5});
        expect(getVisible(wrapper.vm.items)).toEqual([undefined, 3, 4, 5, 6, 7, undefined]);
        wrapper.setData({page: 10});
        expect(getVisible(wrapper.vm.items)).toEqual([undefined, 6, 7, 8, 9, 10]);
    });
});
