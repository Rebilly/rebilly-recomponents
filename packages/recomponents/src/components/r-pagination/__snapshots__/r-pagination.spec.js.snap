// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`r-pagination.vue should render Wrapper and match snapshot 1`] = `
<div>
  <r-pagination-control-stub pagination="[object Object]"></r-pagination-control-stub>
</div>
`;

exports[`r-pagination.vue should render via SSR and match snapshot 1`] = `
<div>
  <div pagination="[object Object]">
    <div class="r-pagination-control">
      <div class="r-pagination-control-buttons-wrapper"><button role="button" disabled="disabled" title="Action is disabled" class="r-button r-pagination-control-button r-button-size-regular r-button-type-default">
          <!----> <svg class="r-icon r-icon-20">
            <use xlink:href="#icon-arrow-left" xmlns:xlink="http://www.w3.org/1999/xlink"></use>
          </svg>
          Prev
        </button> <button role="button" title="" class="r-button r-pagination-control-button r-button-size-regular r-button-type-default">
          <!----> Next<svg class="r-icon r-icon-20">
            <use xlink:href="#icon-arrow-right" xmlns:xlink="http://www.w3.org/1999/xlink"></use>
          </svg></button></div>
    </div>
  </div>
</div>
`;
